File: Examples/test-suite/matlab/constover_runme.m
import constover.*

p = constover.test('test');
if (~strcmp(p,'test'))
    error('test failed~')
end

p = constover.test_pconst('test');
if (~strcmp(p,'test_pconst'))
    error('test_pconst failed~')
end
    
f = constover.Foo();
p = f.test('test');
if (~strcmp(p,'test'))
    error('member-test failed~')
end

p = f.test_pconst('test');
if (~strcmp(p,'test_pconst'))
    error('member-test_pconst failed~')
end

p = f.test_constm('test');
if (~strcmp(p,'test_constmethod'))
    error('member-test_constm failed~')
end

p = f.test_pconstm('test');
if (~strcmp(p,'test_pconstmethod'))
    error('member-test_pconstm failed~')
end

File: Examples/test-suite/javascript/constover_runme.js
var constover = require("constover");

var p = /* await */(constover.test("test"));
if (p != "test") {
    throw "test failed!";
}

p = /* await */(constover.test_pconst("test"));
if (p != "test_pconst") {
    throw "test_pconst failed!";
}

var f = new constover.Foo();

p = /* await */(f.test("test"));
if (p != "test") {
    throw "member-test failed!";
}

p = /* await */(f.test_pconst("test"));
if (p != "test_pconst") {
    throw "member-test_pconst failed!";
}

p = /* await */(f.test_constm("test"));
if (p != "test_constmethod") {
    throw "member-test_constm failed!";
}

p = /* await */(f.test_pconstm("test"));
if (p != "test_pconstmethod") {
    throw "member-test_pconstm failed!";
}
